homeassistant/components/ibeacon/coordinator.py

```diff
@@ -428,11 +462,61 @@ def _async_update_rssi_and_transients(self) -> None:
                     ibeacon_advertisement,
                 )
 
+    @callback
+    def _async_refresh_companion_uuids(self) -> None:
+        """Refresh the list of UUIDs transmitted from the companion app."""
+
+        # When the BLE transmitter is enabled in the companion, the "mobile_app"
+        # integration creates a sensor with unique id suffix "_ble_emitter",
+        # which contains an "id" attribute of the form <uuid>_<major>_<minor>.
+        #
+        self._companion_uuids.clear()
+
+        for entity in self._ent_reg.entities.values():
+            if entity.platform == "mobile_app" and entity.unique_id.endswith(
+                "_ble_emitter"
+            ):
+                state = self.hass.states.get(entity.entity_id)
+                if state and (_id := state.attributes.get("id")):
+                    uuid = _id.split("_")[0]
+                    self._companion_uuids.add(uuid)
+                    self._async_restore_ignored_nameless_ibeacons(uuid)
```

Comment:
This is likely to break in the future since there is no guarentee that the mobile app will continue to use this convention.

If we want to do something like this, we would need to add a published api (in another PR) to the mobile_app integration to get the list of UUIDs instead so we could be sure it keeps working if the mobile_app changes how it registers these in the future.

I'd pull this out of this PR for now since there are multiple new features in this PR which should be two (or more) PRs:
1. allow list for specific uuid
2. mobile_app integration

Reply:
I'm all for clean APIs, but is there someone willing to make a PR on mobile_app and have it merged quickly? (I don't have the cycles) If not I suggest we keep this feature for the following reasons:

- We'd lose a good-enough fix for an ongoing problem (original issue is more than a year old) breaking functionality for many users while waiting potentially another year for the perfect fix.
- We're not doing any crazy hacks with the internals of mobile_app, just reading public entities, the risk of breaking is arguably tolerable
- Even if it breaks in the future, the worst that can happen is to ignore beacons, which is exactly what happens now. So we'd be keeping a currenlty broken system out of fear of having the exact same broken system in the future.

Your call, let me know if you insist on droping this.

Reply:
I think @bdraco is right for this specific part. We can also see translated of renamed entities that would mess with this specific "magic" feature. I would advise to just remove that part, the white listing is already a giant step forward and still relatively easy for end users to voluntary whitelist what they want, and finally track nameless beacons.

Personally, I would even remove the 40004 trick, even if it's already merged on mobile. I think this could cause privacy issues because this means that I can be tracked / or inadvertently track other HA Mobile users when they are around my house. I am not saying this is something new, but it would become more obvious that new HA mobiles are automatically tracked.

Honestly, the basic whitelisting is the perfect and simple solution. We just need that and documentation (web site + in the sensor description) for the fact that BLE Transmitter ID needs to be manually added to HA to track the mobile app.

Reply:
@bdraco : how do you want to proceed?

Btw this PR was meant to replace the 40004 trick, but we could also combine the two if we want a no-config solution for companion users. (Concerning privacy, the real problem is the unique constant UUID, changing the minor doesn't make it more trackable.)

Reply:
Let's take out the mobile_app / _async_refresh_companion_uuids related code here and only do the allow list.  Integration with mobile_app can be added later if an api is merged to the mobile_app.